core/txpool/legacypool/list.go

```diff
@@ -258,6 +258,25 @@ func (m *sortedMap) Flatten() types.Transactions {
 	return txs
 }
 
+// AppendHashes uses the flattened slice of transactions and appends the hashes
+// to the destination slice.
+func (m *sortedMap) AppendHashes(dst []common.Hash) []common.Hash {
```

Comment:
I added this function. It serves as backend for `PendingHashes`, which is used e.g. when we meet another peer on the network, and need to exchange hashes. This happens quite often, so I figured it might be worth optimizing a bit. 
With this change, we just allocate the slice of hashes once, and then we just iterate the flattened cached list and append them. 
The method `Flatten` copies the cache in an intermediate step, to prevent accidental modification. So this allows us to skip that step.